spec.md

```diff
@@ -96,7 +96,51 @@ and eventually have it accepted as a Unicode Technical Standard (UTS).
 
 ## Data Model
 
-> _The data-only representation of a message._
+A Message provides the representation of a single message.
+It takes one of two forms,
+either as a PatternMessage or a SelectMessage.
+
+```ts
+type Message = PatternMessage | SelectMessage
+
+type MessageBody = PatternElement[]
+```
+
+A PatternMessage contains a list of PatternElement values,
+some of which are directly defined literal values,
+while others are placeholders with formatted values that depend on additional data.
+
+```ts
+interface PatternMessage {
+  value: MessageBody
+}
+```
+
+SelectMessage provides for the selection of one list of PatternElement values
+to use as the message's value when formatting,
+depending on the value of one or more Selector values.
+
+Each of the SelectMessage `cases` is defined by a key of one or more string identifiers,
+and selection between them is made according to the corresponding Selector values.
+From this it follows that a valid SelectMessage must have at least as many `select` entries
+as its highest count of string entries within the keys of its `cases`.
```

Comment:
_(Non-blocking)_ I'd suggest to drop the `From this it follows...` sentence.  We'll need to figure this out anyways when specifying the case selection logic, and I don't know if such a requirement will be needed.